FunctionalArray.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  FunctionalArray.swift
+//  SwiftTrainingExercise
+//
+//  Created by Ahmed Ibrahim on 20/06/2023.
+//
+
+import Foundation
+
+class FunctionalArray<Element> where Element: Equatable & Comparable & Hashable {
```

Comment:
why are u using "class"

Reply:
because it have implementation inside of it